### PR TITLE
Initial snippets support

### PR DESCRIPTION
**Pull request by shohamc1**
Adds snippets support for JSim, currently includes snippets for:

- NFET/PFET MOSFETs
- Sub-circuit definition
- Include statements
- Custom components

### CODE REVIEW

**Comment by jamestiotio** [comment]
Thank you for submitting this Pull Request, Shoham. It's going to be really helpful for the SUTD community. :D

Just a suggestion, would it be better to call "custom components" as "user-defined components" instead?

**Comment by shohamc1** [comment]
I believe calling it "custom components" paints a sufficient description of what the snippet is about.

**Comment by jamestiotio** [comment]
Hmm actually right, I think we should align ourselves with MIT’s official JSim documentation.

Based on the documentation, it seems like X should be used for “user-defined subcircuits”.

Also, in that regard, could you kindly assist me in adding snippets for these prefixes as well: R, C, L, V, I and W? You can check the documentation for their official terms. I will include the official MIT documentations for JSim and BSim in this repository to serve as reference for contributors once I get back to my computer.

(I couldn’t find TMSim’s documentation anywhere so if anyone would like to contribute by adding TMSim’s official docs from MIT, that would be great!)

Thank you!

**Comment by shohamc1** [comment]
Latest commit includes the requested changes and additions and also a `.connect` we missed out on.

File: snippets/jsim.json [hunk 1]
```diff
@@ -29,11 +29,60 @@
...
+			"C${1:name} ${2:inputs} ${3:capacitance}"
+		],
+		"description": "Code snippet for a capacitor"
+	},
+	"Inductor": {
+		"prefix": "l",
+		"body": [
+			"L${1:name} ${2:inputs} ${3:inductance}"
+		],
+		"description": "Code snippet for a capacitor"
```
Comment (jamestiotio) [inline snippets/jsim.json:58]: I think you forgot to change this description for the inductor:

```suggestion
		"description": "Code snippet for an inductor"
```

**Comment by jamestiotio** [comment]
By the way, after implementing your changes, do remember to test it locally first before committing.

Also, can I trouble you to raise the semver number to v0.0.2? This is so that the deployment workflow will pass and the latest version will be automatically published to the Visual Studio Code Marketplace.

Thank you!

File: snippets/jsim.json (hunk 1 above)
Comment (shohamc1) [reply snippets/jsim.json:58]: Oops, this is why you should never copy paste code :)

**Comment by shohamc1** [comment]
Fixed the description, updated the README and CHANGELOG.

**Comment by jamestiotio** [comment]
Yay, nice one! This LGTM. Will merge.

**Comment by jamestiotio** [comment]
> 
> 
> (I couldn’t find TMSim’s documentation anywhere so if anyone would like to contribute by adding TMSim’s official docs from MIT, that would be great!)

To give an update on this for future reference, I have found TMSim's documentation and I have committed it with the commit's SHA hash as follows: 29491856420a54c5b5f88ca6b6438724f4c8da2e.

**Pull request merged**